unit-tests/utilities/concurrency/test-usage.cpp

```diff
@@ -0,0 +1,194 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2021 Intel Corporation. All Rights Reserved.
+
+//#cmake:add-file ../../../src/dispatcher.cpp
+
+#include "../../test.h"
+#include <common/utilities/time/timer.h>
+#include <src/concurrency.h>
+
+#include <algorithm>
+#include <vector>
+
+using namespace utilities::time;
+
+// We use this function as a CPU stress test function
+int fibo(int num)
+{
+    if (num < 2) return 1;
+    return fibo(num - 1) + fibo(num - 2);
+}
+
+TEST_CASE( "dequeue wait after stop" )
+{
+    single_consumer_queue< std::function< void( void ) > > scq;
+    timer t( std::chrono::seconds( 1 ) );
+    std::function< void( void ) > f;
+    std::function< void( void ) > * f_ptr = &f;
+
+    scq.enqueue( []() {} );
+    REQUIRE( scq.size() == 1 );
+    REQUIRE( scq.peek( &f_ptr ) );
+
+    REQUIRE( scq.started() );
+    REQUIRE_FALSE( scq.stopped() );
+
+    scq.stop();
+
+    REQUIRE_FALSE( scq.peek( &f_ptr ) );
+    REQUIRE( scq.stopped() );
+    REQUIRE_FALSE( scq.started() );
+
+    REQUIRE( scq.empty() );
+
+    t.start();
+    scq.dequeue( &f, 10000 );
+    REQUIRE_FALSE( t.has_expired() );  // Verify no timeout, dequeue return in less than 10 seconds
```

Comment:
Isn't `t` 1 seconds?

Reply:
Yes it is.
It a timeout of 1 second is not expired it means that the dequeue return false without waiting for 10 seconds.

Reply:
So why do we need a 10-second timeout?

Reply:
We need a timeout that is > 1, we check that if 1 sec timeout did not expired it means that the dequeue didn't end on timeout cause it had a >1 sec timeout

Reply:
I'm asking, then, why we have a 10-second timeout: 2 seconds would be enough to go over 1.

Reply:
No special reason, changed to 2